### ISSUE TITLE
weird output re-scaling if anisotropic input image

### ISSUE THREAD

**Issue by jcohenadad** [neuropoly/spinalcordtoolbox#1074]
data: 

<img width="142" alt="screen shot 2016-12-21 at 19 14 33" src="https://cloud.githubusercontent.com/assets/2482071/21410863/f6353ec0-c7b1-11e6-8eba-f2fe68635383.png">

<img width="57" alt="screen shot 2016-12-21 at 19 13 51" src="https://cloud.githubusercontent.com/assets/2482071/21410865/f90be374-c7b1-11e6-99b9-d8eaddeb5247.png">

**Comment by benjamindeleener** [comment]
NURBS seem to badly approximate the segmentation centerline on this one (see Figure below).
<img width="752" alt="screen shot 2016-12-21 at 22 29 51" src="https://cloud.githubusercontent.com/assets/7757117/21414404/1f2d3b28-c7cd-11e6-87ed-7b53a54ef439.png">

And it is due to edge effects on the segmentation (not anisotropic resolution):
<img width="517" alt="screen shot 2016-12-21 at 22 31 52" src="https://cloud.githubusercontent.com/assets/7757117/21414426/52b8d196-c7cd-11e6-9e14-d3877e4c12a9.png">

Much better when correcting the segmentation:
<img width="752" alt="screen shot 2016-12-21 at 22 33 50" src="https://cloud.githubusercontent.com/assets/7757117/21414464/9c96bec2-c7cd-11e6-8f6c-b52c8b9491ed.png">

Here are the results of straightening:
<img width="287" alt="screen shot 2016-12-21 at 22 42 03" src="https://cloud.githubusercontent.com/assets/7757117/21414594/bcbbc020-c7ce-11e6-82a8-3f431c74220a.png">

**Comment by benjamindeleener** [comment]
We need to find solutions for these edge effects in spinal cord segmentations.

This one can be closed right?

**Comment by jcohenadad** [comment]
yes, we can close this one if we open another one specific to edge effect, with the same priority level (this is a very important issue).

**Comment by jcohenadad** [comment]
@benjamindeleener: if propseg can output a "true" centerline in physical coordinates, would it be possible to compare this centerline with that generated by computing the slicewise center of mass of the generated segmentation, and if it deviates by more than XX mm, then we assume this is caused by a truncated segmentation and we remove the slice.

**Comment by benjamindeleener** [comment]
@jcohenadad good idea. I will look how to implement that. I am working on a similar solution for removing outliers on segmentation edges. I'll try both.

**Comment by jcohenadad** [comment]
actually i was thinking of implementing that as a module in propseg (once the segmentation is spit, it will open it and correct it if necessary).

**Comment by benjamindeleener** [comment]
Yes I was thinking of the same thing

**Comment by jcohenadad** [comment]
didn't you already implement something to solve this issue around June 2016? And maybe it caused issues in very thick slices?

**Comment by benjamindeleener** [comment]
I already implemented this in smooth_centerline function but it is not implemented the same way, and it is not activated all the time. I would like to understand what what is the normal x-y inter-slice distance in a normal population in order to set an appropriate threshold. I will work on that tonight or tomorrow (not home right now)

**Comment by benjamindeleener** [comment]
I implemented what was discussed above:

Removing slices from segmentation when:
1. there are multiple binary objects in the slice
2. the distance between the center of mass of the segmentation and the centerline generated directly by isct_propseg is larger than a threshold (arbitrarily set to 5 mm). The hypothesis here is that the centerline provided by isct_propseg is always at the center of the spinal cord and does not contain edge effect (which should always be the case, as this centerline is the center of the tubular mesh).

Changes have been proposed in PR #1076
